autocti/charge_injection/model/analysis.py

```diff
@@ -63,7 +61,71 @@ def __init__(
             serial_fast_row_lists=serial_fast_row_lists,
         )
 
-    def log_likelihood_function(self, instance: ModelInstance) -> float:
+    def modify_before_fit(self, paths: af.DirectoryPaths, model: af.Collection):
+        """
+        PyAutoFit calls this function immediately before the non-linear search begins, therefore it can be used to
+        perform tasks using the final model parameterization.
+
+        This function:
+
+         1) Visualizes the charge injection imaging dataset, which does not change during the analysis and thus can be
+         done once.
+
+         2) Checks if the noise-map is fixed (it is not if hyper functionality is on), and if it is fixed it
+         sets the noise-normalization to the preloads for computational speed.
+
+        Parameters
+        ----------
+        paths
+            The PyAutoFit paths object which manages all paths, e.g. where the non-linear search outputs are stored,
+            visualization and the pickled objects used by the aggregator output by this function.
+        model
+            The PyAutoFit model object, which includes model components representing the galaxies that are fitted to
+            the imaging data.
+        """
+
+        if not paths.is_complete:
+
+            if not os.environ.get("PYAUTOFIT_TEST_MODE") == "1":
+
+                visualizer = VisualizerImagingCI(visualize_path=paths.image_path)
+
+                visualizer.visualize_imaging_ci(imaging_ci=self.dataset)
+
+                try:
+                    visualizer.visualize_imaging_ci_lines(
+                        imaging_ci=self.dataset, region="parallel_fpr"
+                    )
+                    visualizer.visualize_imaging_ci_lines(
+                        imaging_ci=self.dataset, region="parallel_eper"
+                    )
+
+                except (exc.RegionException, TypeError, ValueError):
+                    pass

```

Comment:
It might be worth putting a logger.debug(e) in here so debugging reports why visualisation did not occur